tests/unit/spec/base/mozilla-fxa-iframe.js

```diff
@@ -0,0 +1,187 @@
+/* For reference read the Jasmine and Sinon docs
+ * Jasmine docs: http://jasmine.github.io/
+ * Sinon docs: http://sinonjs.org/docs/
+ */
+
+/* global describe, beforeEach, afterEach, it, expect */
+
+describe('mozilla-fxa-iframe.js', function() {
+
+    'use strict';
+
+    var fxaHost = 'http://localhost:9876';
+
+    beforeEach(function () {
+        var fxaMarkup = [
+            '<section id="fxa-iframe-config" data-host="' + fxaHost + '">' +
+                '<iframe id="fxa" data-src="' + fxaHost + '?utm_campaign=fxa-embedded-form&amp;utm_medium=referral&amp;utm_source=firstrun&amp;utm_content=fx-{{ version }}&amp;entrypoint=firstrun&amp;service=sync&amp;context=iframe&amp;style=chromeless&amp;haltAfterSignIn=true"></iframe>' +
+            '</section>'
+        ].join();
+
+        $(fxaMarkup).appendTo($('body'));
+    });
+
+    afterEach(function () {
+        $('#fxa-iframe-config').remove();
+    });
+
+    describe('Mozilla.FxaIframe.init', function () {
+        it('should initialize the FxA iframe', function () {
+            Mozilla.FxaIframe.init();
+
+            // make sure iframe src is as expected
+            expect($('#fxa').attr('src')).toContain(fxaHost);
+        });
+
+        it('should update iframe src for Fx 46+', function () {
+            var fxVersion = Mozilla.Client.FirefoxMajorVersion;
+            Mozilla.Client.FirefoxMajorVersion = 47;
+
+            Mozilla.FxaIframe.init();
+
+            expect($('#fxa').attr('src')).toContain('context=fx_firstrun_v2');
+
+            Mozilla.Client.FirefoxMajorVersion = fxVersion;
+        });
+
+        it('should not update iframe src for Fx < 46', function () {
+            var fxVersion = Mozilla.Client.FirefoxMajorVersion;
+            Mozilla.Client.FirefoxMajorVersion = 45;
+
+            Mozilla.FxaIframe.init();
+
+            expect($('#fxa').attr('src')).not.toContain('context=fx_firstrun_v2');
+
+            Mozilla.Client.FirefoxMajorVersion = fxVersion;
+        });
+
+        it('should append URL encoded email if sent in config', function () {
```

Comment:
Would it be worth adding another test here to make sure invalid email addresses aren't encoded in the URL?